### PR TITLE
Force update resource sets

### PR DESCRIPTION
**Pull request by g-gaston**
*Description of changes:*
 Builds on top of #526 and implements the first part: re-applying the objects managed by the `ClusterResourceSet`
This should take care of vsphere `syncer` and `driver` images, which prior to this didn't ever get updated

I added comments on the code explaining why each action is needed and what should be the long term solution

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/532#" title="Author self-approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (danbudris) [inline pkg/providers/vsphere/vsphere.go:1591]: should we have a task for this?

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/532#pullrequestreview-794653340" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/532#" title="Author self-approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (g-gaston) [reply pkg/providers/vsphere/vsphere.go:1591]: https://github.com/kubernetes-sigs/cluster-api/issues/5555

**Comment by abhay-krishna** [comment]
@g-gaston: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-e2e-presubmit/1455322687466377216) | `/test eks-anywhere-e2e-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=532). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:g-gaston).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@g-gaston: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-e2e-presubmit/1455322687466377216) | `/test eks-anywhere-e2e-presubmit`
eks-anywhere-attribution-files-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-attribution-files-presubmit/1455322687424434176) | `/test eks-anywhere-attribution-files-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=532). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:g-gaston).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@g-gaston: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-e2e-presubmit/1455322687466377216) | `/test eks-anywhere-e2e-presubmit`
eks-anywhere-attribution-files-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-attribution-files-presubmit/1455322687424434176) | `/test eks-anywhere-attribution-files-presubmit`
eks-anywhere-mocks-presubmits | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-mocks-presubmits/1455322687512514560) | `/test eks-anywhere-mocks-presubmits`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=532). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:g-gaston).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@g-gaston: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-e2e-presubmit/1455322687466377216) | `/test eks-anywhere-e2e-presubmit`
eks-anywhere-attribution-files-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-attribution-files-presubmit/1455322687424434176) | `/test eks-anywhere-attribution-files-presubmit`
eks-anywhere-mocks-presubmits | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-mocks-presubmits/1455322687512514560) | `/test eks-anywhere-mocks-presubmits`
eks-anywhere-presubmit | fa3d18f2265951280c8ef190440de850f50a237e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/532/eks-anywhere-presubmit/1455322687550263296) | `/test eks-anywhere-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=532). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:g-gaston).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Pull request merged**
